base/dict.jl

```diff
@@ -392,6 +392,47 @@ function setindex!(h::Dict{K,V}, v0, key::K) where V where K
     return h
 end
 
+function modify!(f, h::Dict{K, V}, key0) where {K, V}
+    key = convert(K, key0)
+    if !isequal(key, key0)
+        throw(ArgumentError("$(limitrepr(key0)) is not a valid key for type $K"))
+    end
+
+    # Ideally, to improve performance for the case that requires
+    # resizing, we should use something like `ht_keyindex` while
+    # keeping computed hash value and then do something like
+    # `ht_keyindex2!` if `f` returns non-`nothing`.
+    idx = ht_keyindex2!(h, key)
+
+    age0 = h.age
+    if idx > 0
+        @inbounds vold = h.vals[idx]
+        vold = Some{V}(vold)
+        vnew = f(vold)
```

Comment:
In the abstract dict case we use `old` and `val` here we use `vold` and `vnew` and in the documentation we use `old` and `new`. We should probably be consistent and I prefer `old` and `new`